src/apps/tools.clj

```diff
@@ -96,19 +108,24 @@
   [{:keys [tools]}]
   (transaction
     (let [tool-ids (doall (map add-new-tool tools))]
+      (dorun (map permissions/register-public-tool tool-ids))
```

Comment:
Since this is inside a `transaction`, I think this could run the risk of leaving orphaned entries in the permissions service, if something goes wrong while adding these permissions. But at least it won't attempt to add any permissions if something goes wrong while adding one of the tools.